### PR TITLE
add `dagger mod publish`

### PR DESCRIPTION
**Pull request by vito**
Pretty rough start, doesn't complain about errors at all, and if you publish before pushing, Daggerverse currently caches the error forever (until it reboots). Whoopsie!

Here's how it looks:

[asciicast](https://asciinema.org/a/LtDT5YB9ogyHdoShgbEDX29UV)

### CODE REVIEW

**Review by gerhard — APPROVED** [review]
💪 🚀

Want to merge this @sipsma if it looks good to you too?

**Review by sipsma — APPROVED** [review]
LGTM, tested successfully locally https://daggerverse.dev/mod/github.com/sipsma/daggerverse/testpublish@e2377da1ddf53f2263b7e5c1e986d5c3ad57d839

Will wait to see if @jpadams has any feedback before merging

**Review by jpadams — APPROVED** [review]
LGTM!
Let's ship it and improve things from there.

Limitations/caveats today:
- needs to be run from within the module directory or with `-m <local dir path>`
  - This is a bit confusing today because the help text for global `-m` suggests that `git` refs are also acceptable, which they aren't in this case.
  
- depends on a git remote named `origin`

**Pull request merged**
